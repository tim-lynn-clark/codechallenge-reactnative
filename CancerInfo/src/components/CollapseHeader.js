import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';

import Ionicons from 'react-native-vector-icons/Ionicons';
import {Colors} from 'react-native/Libraries/NewAppScreen';

import Color from '../Colors';
/**
 * @author Tom Jay
 * @function CollapseHeader
 **/
const CollapseHeader = ({collapsed, onPress, title}) => {
  return (
    <View style={styles.headerContainer}>
      <View style={styles.headerTextContainer}>
        <Text style={styles.headerTitle}>{title}</Text>
      </View>

      <TouchableOpacity
        onPress={onPress}
        style={styles.expandCollapseButtonContainer}>
        {collapsed && (
          <Ionicons name="chevron-down-outline" size={25} color="gray" />
        )}
        {!collapsed && (
          <Ionicons name="chevron-up-outline" size={25} color="gray" />
        )}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerContainer: {
    marginTop: 20,
    marginLeft: 20,
    flexDirection: 'row',
    alignContent: 'center',
  },
  headerTextContainer: {
    flexDirection: 'column',
    backgroundColor: '#b3d4fc',
    justifyContent: 'center',
  },
  headerTitle: {
    marginLeft: 5,
    width: 300,
    fontSize: 18,
    fontWeight: 'bold',
    color: Colors.black,
  },
  expandCollapseButtonContainer: {
    backgroundColor: Colors.ltBlue,
    paddingRight: 5,
  },
});

export default CollapseHeader;
