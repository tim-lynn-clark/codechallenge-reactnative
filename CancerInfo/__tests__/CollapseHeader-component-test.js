import React from 'react';
import {cleanup, render} from '@testing-library/react-native';

import CollapseHeader from '../src/components/CollapseHeader';

afterEach(cleanup);

describe('CollapseHeader', () => {
  it('CollapseHeader should render', () => {
    //   const helloWorldText = 'Hello World!';

    const {toJSON, getByText} = render(<CollapseHeader />);

    //    const foundHelloWorldText = getByText(helloWorldText);

    //   expect(foundHelloWorldText.props.children).toEqual(helloWorldText);
    expect(toJSON()).toMatchSnapshot();
  });
});
