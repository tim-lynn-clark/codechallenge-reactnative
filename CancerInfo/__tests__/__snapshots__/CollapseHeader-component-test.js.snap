// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseHeader CollapseHeader should render 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "marginLeft": 20,
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#b3d4fc",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginLeft": 5,
          "width": 300,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "opacity": 1,
        "paddingRight": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gray",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ïˆž
    </Text>
  </View>
</View>
`;
